Test: Update Vagrant box to version 83

eloycoto opened the pull request:
Updated Vagrant box to version 83, and fix the issue with CNI.

Fix #4429

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/4432)
<!-- Reviewable:end -->

eloycoto committed:
Test: Update Vagrant box to version 83

Updated Vagrant box to version 83, and fix the issue with CNI.

Fix #4429

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

ianvernon commented on test/runtime/connectivity.go:
```diff
@@ -188,8 +188,23 @@ var _ = Describe("RuntimeValidatedConnectivityTest", func() {
+		BeforeAll(func() {
+			// Remove any CNI plugin installed in the provision server. This
+			// helps to avoid issues on installing the new CNI
+			_ = vm.ExecWithSudo(fmt.Sprintf("rm -rf %[1]s/*.conf", netDPath)).ExpectSuccess(
```
Ideally it shouldn't, but does this affect cri-o support within the VM after the test is ran? Furthermore, if users run this locally (which is a supported option as specified in the Ginkgo contributing guides), we may be removing configuration files that users have installed locally.

eloycoto committed:
Revert "Revert to working Vagrant box and cilium-builder image"

This reverts commit 00d75f00fe51e37df61d4fab3ad4def2dced2850.

rlenglet reviewed (approved):
LGTM, thanks!

eloycoto committed:
Test: provision uninstall crio cni interface

Disabled default file `/etc/cni/net.d/100-crio-bridge.conf` to avoid
issues where kube-dns uses crio CNI plugin instead of cilium one.

Fail example:
```
cmd: kubectl get pods -o wide --all-namespaces
NAMESPACE     NAME                           READY     STATUS    RESTARTS   AGE       IP              NODE
kube-system   cilium-ccm6b                   1/1       Running   0          1m        192.168.36.12   k8s2
kube-system   cilium-qcvpr                   1/1       Running   0          1m        192.168.36.11   k8s1
kube-system   etcd-k8s1                      1/1       Running   0          8m        192.168.36.11   k8s1
kube-system   kube-apiserver-k8s1            1/1       Running   0          8m        192.168.36.11   k8s1
kube-system   kube-controller-manager-k8s1   1/1       Running   0          9m        192.168.36.11   k8s1
kube-system   kube-dns-59d8c5f9b5-p9rfn      3/3       Running   0          8m        10.88.0.2       k8s1
kube-system   kube-proxy-68gn6               1/1       Running   0          1m        192.168.36.12   k8s2
kube-system   kube-proxy-9wggj               1/1       Running   0          8m        192.168.36.11   k8s1
kube-system   kube-scheduler-k8s1            1/1       Running   0          9m        192.168.36.11   k8s1
```

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>
